pkg/apis/alibabacloudprovider/v1beta1/alibabacloudmachineproviderconfig_types.go

```diff
@@ -247,6 +179,177 @@ type AlibabaCloudMachineProviderConfigList struct {
 	Items           []AlibabaCloudMachineProviderConfig `json:"items"`
 }
 
+// InstanceTenancy Specifies whether to create the instance on a dedicated host
+type InstanceTenancy string
+
+type SystemDiskProperties struct {
+	//The category of the system disk. Valid values:
+	//cloud_essd: ESSD. When the parameter is set to this value, you can use the SystemDisk.PerformanceLevel parameter to specify the performance level of the disk.
+	//cloud_efficiency: ultra disk.
+	//cloud_ssd: standard SSD.
+	//cloud: basic disk.
+	//For non-I/O optimized instances of retired instance types, the default value is cloud. For other instances, the default value is cloud_efficiency.
+	// +optional
+	Category string `json:"category,omitempty"`
+
+	//The performance level of the ESSD used as the system disk. Default value: PL1. Valid values:
+	//PL0: A single ESSD can deliver up to 10,000 random read/write IOPS.
+	//PL1: A single ESSD can deliver up to 50,000 random read/write IOPS.
+	//PL2: A single ESSD can deliver up to 100,000 random read/write IOPS.
+	//PL3: A single ESSD can deliver up to 1,000,000 random read/write IOPS.
+	//For more information about ESSD performance levels, see ESSDs.
+	// +optional
+	PerformanceLevel string `json:"performanceLevel,omitempty"`
+
+	//The name of the system disk. The name must be 2 to 128 characters in length. It must start with a letter and cannot start with http:// or https://. It can contain letters, digits, colons (:), underscores (_), and hyphens (-).
+	//This parameter is empty by default.
+	// +optional
+	Name string `json:"name,omitempty"`
+
+	//The size of the system disk. Unit: GiB. Valid values: 20 to 500.
+	//The value must be at least 20 and greater than or equal to the size of the image.
+	//The default value is 40 or the size of the image, depending on whichever is greater.
+	// +optional
+	Size int `json:"size,omitempty"`
+
+	//The description of the system disk. The description must be 2 to 256 characters in length and cannot start with http:// or https://.
+	//This parameter is empty by default.
+	// +optional
+	Description string `json:"description,omitempty"`
+}
+
+// DataDisk The datadisk of Instance
+type DataDisk struct {
```

Comment:
Wondering if a `SystemDisk` is actually a `DataDisk` as well, are all of these params applicable to both?

Reply:
It appears that the `SystemDisk` and the `DataDisk` types have different properties.
https://github.com/aliyun/alibaba-cloud-sdk-go/blob/1c273f0a7d673a8f941f67cfe572c0a3ea1a7122/services/ecs/struct_data_disk.go
https://github.com/aliyun/alibaba-cloud-sdk-go/blob/64043b70c5e4a928281b4114300481e027111abb/services/ens/struct_system_disk.go